### DISCUSSION TITLE
Why is the class name underlined?

### DISCUSSION THREAD

**Discussion by navneethc** [sharkdp/bat#2453] [General]
I opened a Python file with a class in it and found that class name was underlined when I opened it with bat. Why is that and what can I do to make it go away? 

![image](https://user-images.githubusercontent.com/4192839/213556360-de23ea3c-6049-4129-9895-b408b306a032.png)

I was about to create an issue when I ran into #2110 where the user mentions an underline feature that was added. I couldn't find more information about that. 

Ubuntu 20.04 via Windows 10/WSL 2
Windows Terminal  1.15.3465.0
bat 0.22.1
Fish shell 3.6.0
Starship 1.9.1

I alias `cat` to `bat -n`.

**Comment by keith-hall** [comment]
This is down to the color scheme (aka theme) used - they can make any text bold, italic or underlined. I believe your options are to configure Bat to use a different theme, or to edit the theme to remove the underline styles and rebuild the bat cache.

**Reply by navneethc** [reply]
Thanks for the helpful reply, @keith-hall. Is there a reason why themes are allowed to underline or, in general, change apparently random words without the user's input? (Happy to be directed to the appropriate document. :) )

**Reply by keith-hall** [reply]
I think to find documentation on it, we'd have to search for some old TextMate docs, if such exists, as I believe this option was introduced there. I would guess the reason was to support underlining links etc. But I'm not sure why some themes abuse it to underline class names, I guess whomever designed the theme wanted such parts to stand out without wanting them bold, but this is pure speculation.

**Reply by sharkdp** [reply]
For italic text, we have an option in `bat` to enable/disable it:
```
      --italic-text <when>
          Specify when to use ANSI sequences for italic text in the output. Possible values: always,
          *never*.
```

Maybe we should consider doing something similar for underlined text? This way, underlining could be disabled globally.
